mantidproject/mantid
# Updated errorreporter to correctly specify application name

**Description of work.**

This updates the error-reporter so that it sends the correct aplication name i.e. workbench or mantidplot

This PR also updates the dialog to not refer specifically to mantidplot

**Report to:** @martyngigg  

**To test:**

Unfortunately I think this needs to be tested in packaged versions of linux, OSX and windows builds as the launch scripts for all of them have been changed. For each the following should be checked.

* For workbench and mantidplot check that the errorreport window appears when you run the segfault algorithm. (To run this in workbench run Segfault(False) in the script interpreter).
* Check that the resulting error message contains the correct application name. This can be done either by leaving mantid as it is and checking the errorreport server for the report or changeing the errorreport url in the properties file to something like https://ptsv2.com/s/howitworks.html and checking the body of the message contains the correct application name.

* For workbench and mantidplot check the same things whilst provoking an uncaught exception. Check that the dialog is now non specific
* For mantidplot running `ws = CreateWorkspace(DataX=[0, 1], DataY=[float('inf'), float('inf')])` and trying to plot the resulting workspace should do this.
* For workbench add `raise` before line 88 in workspacewidget.py and then try plotting anything



Fixes #25262  


*This does not require release notes* because it is not a user facing issue


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
